ios/packages/core/Sources/utilities/JSValue+Extensions.swift

```diff
@@ -0,0 +1,78 @@
+//
+//  JSValue+Extensions.swift
+//  PlayerUI
+//
+//  Created by Zhao Xia Wu on 2024-01-18.
+//
+
+import Foundation
+import JavaScriptCore
+
+extension JSValue {
+    var tryCatchWrapper: JSValue? {
+        self.context.evaluateScript(
+        """
+           (fn, args) => {
+               try {
+                   console.log(args)
+                   return fn(...args)
+               } catch(e) {
+                   console.log(e)
+                   if (e instanceof Error) {
+
+                      return e
+                }
+             }
+           }
+        """)
+
+    }
+
+    var errorCheckWrapper: JSValue? {
+        self.context.evaluateScript(
+        """
+                (obj) => (obj instanceof Error)
+        """)
+    }
+
+
+    /**
+     A way to catch errors for void functions not called inside a player process. Called on void functions.
+     - parameters:
+        - args: List of arguments taken by the function
+     */
+    public func callTryCatchWrapper(args: Any...) throws {
+        let result = self.tryCatchWrapper?.call(withArguments: [self, args])
```

Comment:
Since `tryCatchWrapper` is a computed property, that means it evals that snippet every time we call this

does JavascriptCore clean it up automatically correctly? or should will we need to handle something?